consensus/src/liveness/proposal_generator.rs

```diff
@@ -156,11 +216,29 @@ impl ProposalGenerator {
             // the local time exceeds it.
             let timestamp = self.time_service.get_current_timestamp();
 
+            let (max_block_txns, max_block_bytes) = if let Some(value) = chain_health_backoff {
+                let max_block_txns = self
+                    .max_block_txns
+                    .min(value.max_sending_block_txns_override);
+                let max_block_bytes = self
+                    .max_block_bytes
+                    .min(value.max_sending_block_bytes_override);
+
+                warn!(
+                    "Generating proposal reducing limits to {} txns and {} bytes, due to chain health backoff",
```

Comment:
Aside from the log, should we also have a counter on how many times each chain_health_backoff was triggered?